pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamConfig.java

```diff
@@ -16,172 +16,207 @@
-
-    // TODO: Move zkBrokerURL and bootstrapHosts to kafka specific config objects
-    _zkBrokerUrl = streamConfigMap.get(KafkaStreamConfigProperties.constructStreamProperty(
-        KafkaStreamConfigProperties.HighLevelConsumer.KAFKA_HLC_ZK_CONNECTION_STRING));
-
-    final String bootstrapHostConfigKey = KafkaStreamConfigProperties.constructStreamProperty(
-        KafkaStreamConfigProperties.LowLevelConsumer.KAFKA_BROKER_LIST);
-    if (streamConfigMap.containsKey(bootstrapHostConfigKey)) {
-      _bootstrapHosts = streamConfigMap.get(bootstrapHostConfigKey);
-    } else {
-      _bootstrapHosts = null;
-    }
 
-    String consumerTypesCsv = streamConfigMap.get(
-        StreamConfigProperties.constructStreamProperty(_streamType, StreamConfigProperties.STREAM_CONSUMER_TYPES));
-    Iterable<String> parts = Splitter.on(',').trimResults().split(consumerTypesCsv);
-    for (String part : parts) {
-      _consumerTypes.add(ConsumerType.valueOf(part.toUpperCase()));
-    }
-    if (_consumerTypes.isEmpty()) {
-      throw new RuntimeException("Empty consumer types: Must have 'highLevel' or 'simple'");
+    _type = streamConfigMap.get(StreamConfigProperties.STREAM_TYPE);
+    Preconditions.checkNotNull(_type, "Stream type cannot be null");
+
+    String topicNameKey =
+        StreamConfigProperties.constructStreamProperty(_type, StreamConfigProperties.STREAM_TOPIC_NAME);
+    _topicName = streamConfigMap.get(topicNameKey);
+    Preconditions.checkNotNull(_topicName, "Stream topic name " + topicNameKey + " cannot be null");
+
+    String consumerTypesKey =
+        StreamConfigProperties.constructStreamProperty(_type, StreamConfigProperties.STREAM_CONSUMER_TYPES);
+    String consumerTypes = streamConfigMap.get(consumerTypesKey);
+    Preconditions.checkNotNull(consumerTypes, "Must specify at least one consumer type " + consumerTypesKey);
+    for (String consumerType : consumerTypes.split(",")) {
+      if (consumerType.equals(SIMPLE_CONSUMER_TYPE_STRING)) { //For backward compatibility of stream configs which referred to lowlevel as simple
+        consumerType = ConsumerType.LOWLEVEL.toString();
+      }
+      _consumerTypes.add(ConsumerType.valueOf(consumerType.toUpperCase()));
     }
-    Collections.sort(_consumerTypes);
 
-    String offsetCriteriaProperty = StreamConfigProperties.constructStreamProperty(_streamType, StreamConfigProperties.STREAM_CONSUMER_OFFSET_CRITERIA);
-    String offsetCriteria = streamConfigMap.get(offsetCriteriaProperty);
-    if (offsetCriteria == null) {
-      _offsetCriteria = DEFAULT_OFFSET_CRITERIA;
-    } else {
+    String consumerFactoryClassKey =
+        StreamConfigProperties.constructStreamProperty(_type, StreamConfigProperties.STREAM_CONSUMER_FACTORY_CLASS);
+    _consumerFactoryClassName = streamConfigMap.get(consumerFactoryClassKey);
+    Preconditions.checkNotNull(_consumerFactoryClassName,
+        "Must specify consumer factory class name in property " + consumerFactoryClassKey);
```

Comment:
Hi, we are getting backward-incompatible issue for this. "consumer.factory.class.name" is not a required config for old realtime tables. Could we infer this value?

Below are typical configs are are using:
For highlevel consumer:
```
      "streamConfigs": {
        "stream.kafka.decoder.class.name": "com.uber.uPinot.decoder.HeatpipeKafkaMessageDecoder",
        "streamType": "kafka",
        "stream.kafka.hlc.zk.connect.string": "kafka-zookeeper",
        "realtime.segment.flush.threshold.size": "1000000",
        "stream.kafka.consumer.type": "highLevel",
        "realtime.segment.flush.threshold.time": "18000000",
        "stream.kafka.consumer.prop.auto.offset.reset": "smallest",
        "stream.kafka.topic.name": "some-topic"
      },
```
For simple consumer
```
      "streamConfigs": {
        "stream.kafka.decoder.class.name": "com.uber.uPinot.decoder.HeatpipeKafkaMessageDecoder",
        "streamType": "kafka",
        "stream.kafka.hlc.zk.connect.string": "kafka-zookeeper",
        "realtime.segment.flush.threshold.size": "1000000",
        "stream.kafka.consumer.type": "simple",
        "realtime.segment.flush.threshold.time": "18000000",
        "stream.kafka.broker.list": "kafka-broker",
        "stream.kafka.consumer.prop.auto.offset.reset": "largest",
        "stream.kafka.topic.name": "sometopic"
      }
```